app/components/Schema/Dropdown/index.js

```diff
@@ -20,90 +20,105 @@
  */
 
 import React, { Component } from 'react';
-import { FormGroup, Input } from 'reactstrap';
-import type { specImplementationPropType } from '../../../utils/sharedTypes';
+import Select from 'react-select';
 
 type Props = {
-  index: string,
-  columns: Array<string>,
-  stream: { metadata?: Array<{}>, replication_key?: string },
-  specImplementation: specImplementationPropType,
-  handleChange: (field: string, index: string, value: boolean | string) => void
+  editField: boolean,
+  placeholder: string,
+  values: Array<string>,
+  defaultValues: Array<string>,
+  isMulti: boolean,
+  streamMetadata?: { index?: number, metadata?: {} },
+  index: number,
+  handleChange: () => void,
+  field: string
+};
+
+const colourStyles = {
+  option: (styles, { isFocused }) => ({
+    ...styles,
+    backgroundColor: isFocused ? '#dbd5ff' : null
+  }),
+  multiValue: (styles, { isDisabled }) => ({
+    ...styles,
+    backgroundColor: isDisabled ? 'hsl(0,0%,90%)' : '#5c56a5',
+    color: isDisabled ? '#000' : '#fff'
+  }),
+  multiValueLabel: (styles, { isDisabled }) => ({
+    ...styles,
+    color: isDisabled ? '#000' : '#fff'
+  })
 };
 
 export default class Dropdown extends Component<Props> {
-  handleChange = (e: SyntheticEvent<HTMLButtonElement>) => {
-    const { value } = e.currentTarget;
-    this.props.handleChange('replication-key', this.props.index, value);
-  };
+  constructor(props) {
+    super(props);
 
-  getReplicationKey(
-    stream: {
-      metadata: Array<{ breadcrumb: ?Array<string> }>,
-      replication_key?: string,
-      breadcrumb?: Array<string>
-    },
-    specImplementation?: {} = {}
-  ) {
-    const { replicationKey: repKeyMetadata = true } =
-      specImplementation.usesMetadata || {};
-    if (!repKeyMetadata) {
-      return stream.replication_key;
-    }
+    const { defaultValues, editField } = props;
 
-    let indexToUpdate;
+    this.state = {
+      selectedOptions: defaultValues.map((value) => ({
+        value,
+        label: value
+      })),
+      editField
+    };
+  }
 
-    // Look for the metadata with the empty breadcrumb
-    stream.metadata.forEach((metadata, index) => {
-      if (metadata.breadcrumb.length === 0) {
-        indexToUpdate = index;
-      }
-    });
+  getOptions = () => {
+    const { values } = this.props;
 
-    if (indexToUpdate !== undefined) {
-      const replicationKey = this.props.stream.metadata[indexToUpdate].metadata[
-        'replication-key'
-      ];
+    return values.map((value) => ({
+      value,
+      label: value
+    }));
+  };
 
-      return replicationKey;
-    }
+  handleChange = (selectedOptions) => {
+    const { field } = this.props;
 
-    return '';
-  }
+    if (field === 'keyFields') {
```

Comment:
This is a "smell" in my opinion. In other words, it's proof that this generic Dropdown component isn't generic after all. Largely, I don't see why we need our own select component when it's so similar to the Select component in `react-select`. I don't think we need to spend any more time on it, but largely the approach to reuse here did not work.